gems/pending/middleware/events/middleware_event_monitor.rb

```diff
@@ -0,0 +1,31 @@
+class MiddlewareEventMonitor
+  def initialize(ems, start_time)
+    @ems = ems
+    @alerts_client = ems.alerts_connect
+    # TODO: can we store and resume where we left off?  To avoid duplicates, only look for events now or later,
+    # so basically we miss events when we're not running...
+    @start_time = start_time || Time.now.to_i * 1000
+  end
+
+  def start
+    # nothing to do, not an active connection, just making rest calls
+  end
+
+  def stop
+    # nothing to do, not an active connection, just making rest calls
+  end
+
+  def each
+    # Just fetch up to the current time, then rest current time as the next start time
+    end_time = Time.now.to_i * 1000
+    $mw_log.debug("Catching Events for interval [Start=#{@start_time}, End=#{end_time}]")
+
+    @alerts_client.list_events("startTime" => @start_time, "endTime" => end_time).each do |event|
+      yield event
```

Comment:
After looking back at this I think the confusion is the method name `each`.  That would imply an outer loop, but there really isn't one.  It's just a fetch of new events for the incrementing time range.  The block should only be executed for the events returned by `list_events`.  I'll rename from `each` to `monitor`.